### PR TITLE
Fix multi-line range formatting when on first line

### LINKED ISSUE #1244: When pasting at the start of  a multiline string pipe is added unnecessarily
**Describe the bug**
When pasting in a multiline string with existing `|` a new `|` is added before the string itself/

**To Reproduce**
1.Make sure `On paste formatting` is enabled
2.Create multiline string like:
```
s"""| this is@@ a multiline
      | string
      |""".stripMargin
```
3. Paste something not multiline at the position of `@@`
4. `|` will be added before the `s"""`

**Expected behavior**
Nothing should be added there.

**Screenshots**

image

**Installation:**
- Operating system: Linux
- Editor: Visual Studio Code
- Metals version: newest snapshot

**Search terms**
multiline on paste formatting pipe

### PR DESCRIPTION
**Pull request by ckipp01**
Closes #1244 

When I originally worked on this (my first Metals ticket 🎉 ), I totally forgot about the case where you paste on the first line of the multi-line string. This adds in a check for that and also a test for it.

### CODE REVIEW

**Comment by ckipp01** [comment]
``` 
[error] Failed tests:
[error] 	tests.BspSwitchLspSuite`
```
Test failure seems to be unrelated.

File: metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala [hunk 1]
```diff
@@ -159,14 +159,24 @@ final class MultilineStringFormattingProvider(
...
             val pipePos = new Position(line, firstPipeIndex)
-            new TextEdit(new Range(zeroPos, pipePos), defaultIndent)
+            val textEdit =
+              new TextEdit(new Range(zeroPos, pipePos), defaultIndent)
+            Some(textEdit)
         }
       case _ =>
-        val newText = defaultIndent + "|"
-        new TextEdit(new Range(zeroPos, zeroPos), newText)
+        val isFirstLineOfMultiLine = lineText.trim.contains("\"\"\"|")
```
Comment (olafurpg) [inline metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala]: What happens if the line doesn’t have a pipe? It’s optional

Comment (ckipp01) [reply metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala]: If there is no pipe on the line it gets filtered out beforehand and the line never enters the `formatPipeLine` method. I suppose I should have added a test for this as well. I did just check it locally and that indeed is how it works.

Comment (ckipp01) [reply metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala]: I just added another test for this https://github.com/scalameta/metals/pull/1249

**Pull request merged**